### PR TITLE
Add utility methods for Futures

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public final class MoreFutures
+{
+    private MoreFutures()
```
Comment (electrum) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: Use `{}`

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        return future;
+    }
+
+    public static <V> CompletableFuture<V> toCompletableFuture(ListenableFuture<V> listenableFuture)
+    {
+        requireNonNull(listenableFuture, "listenableFuture is null");
+
+        CompletableFuture<V> future = new CompletableFuture<>();
+        listenableFuture.addListener(() -> {
+                    try {
```
Comment (electrum) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: This formatting looks off

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                    catch (InterruptedException e) {
+                        // this should not happen
+                        future.completeExceptionally(e);
+                    }
+                    catch (ExecutionException e) {
+                        Throwable cause = e.getCause() == null ? e : e.getCause();
+                        future.completeExceptionally(cause);
+                    }
+                },
+                MoreExecutors.directExecutor());
```
Comment (electrum) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: Static import

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        requireNonNull(listenableFuture, "listenableFuture is null");
+
+        CompletableFuture<V> future = new CompletableFuture<>();
+        listenableFuture.addListener(() -> {
+                    try {
+                        future.complete(listenableFuture.get());
+                    }
+                    catch (InterruptedException e) {
+                        // this should not happen
+                        future.completeExceptionally(e);
```
Comment (electrum) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: I think this should interrupt the thread

**Comment by electrum** [comment]
Since these should all be used with static import, why not name it `Futures`?

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                if (exception != null) {
+                    future.completeExceptionally(exception);
+                }
+                else {
+                    future.complete(value);
+                }
+            });
+            hasFuture = true;
+        }
+        checkArgument(hasFuture, "futures is empty");
```
Comment (erichwang) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: You can just do checkArgument(Iterables.isEmpty(futures), ...) at the top

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        return future;
+    }
+
+    public static <V> CompletableFuture<V> toCompletableFuture(ListenableFuture<V> listenableFuture)
+    {
+        requireNonNull(listenableFuture, "listenableFuture is null");
+
+        CompletableFuture<V> future = new CompletableFuture<>();
+        listenableFuture.addListener(() -> {
```
Comment (erichwang) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: Why don't you use the addCompletionCallback to the listenable future?

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                    future.complete(value);
+                }
+            });
+            hasFuture = true;
+        }
+        checkArgument(hasFuture, "futures is empty");
+        return future;
+    }
+
+    public static <V> CompletableFuture<V> toCompletableFuture(ListenableFuture<V> listenableFuture)
```
Comment (erichwang) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: make sure to propagate cancellation to the listenableFuture

File: concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+                        Throwable cause = e.getCause() == null ? e : e.getCause();
+                        future.completeExceptionally(cause);
+                    }
+                },
+                MoreExecutors.directExecutor());
+
+        return future;
+    }
+
+    public static <V> ListenableFuture<V> toListenableFuture(CompletableFuture<V> completableFuture)
```
Comment (erichwang) [inline concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java]: make sure to propagate cancellation to the completableFuture

**Pull request merged**